### ISSUE TITLE
cascade extension applied twice (Trac #1763)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1755]
From slack:

  @musner just presented on the cascade call that there is an issue in simprod with cascade simulation. Short summary is: cascasde extension is simulated twice, first by CMC, then by clsim. So it needs to be disabled in clsim. I added an option to
  simprod-scripts/python/segments/HybridPhotonicsCLSim.py / the PropagatePhotons segment needs to be used with UseCascadeExtension = False when the PropagateMuons() segment is used in the chain before it.

  I think this needs to be added somewhere in
  simprod-scripts/python/modules/*.py, but I am unsure where

  it also must not be set if you don’t use CMC (i.e. for Genie/Geant4 I guess)

  could you add the relevant options to the various scripts? We could also make
  UseCascadeExtension = False the default

This requires a bugfix release of simulation.

Migrated from https://code.icecube.wisc.edu/ticket/1763
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:12:47", 
    "description": "From slack:\n\n  @musner just presented on the cascade call that there is an issue in simprod with cascade simulation. Short summary is: cascasde extension is simulated twice, first by CMC, then by clsim. So it needs to be disabled in clsim. I added an option to\n  simprod-scripts/python/segments/HybridPhotonicsCLSim.py / the PropagatePhotons segment needs to be used with UseCascadeExtension = False when the PropagateMuons() segment is used in the chain before it.\n\n  I think this needs to be added somewhere in\n  simprod-scripts/python/modules/*.py, but I am unsure where\n\n  it also must not be set if you don\u2019t use CMC (i.e. for Genie/Geant4 I guess)\n\n  could you add the relevant options to the various scripts? We could also make\n  UseCascadeExtension = False the default\n\nThis requires a bugfix release of simulation.", 
    "reporter": "claudio.kopper", 
    "cc": "olivas", 
    "resolution": "fixed", 
    "_ts": "1550067167842669", 
    "component": "combo simulation", 
    "summary": "cascade extension applied twice", 
    "priority": "blocker", 
    "keywords": "", 
    "time": "2016-06-27T18:53:20", 
    "milestone": "", 
    "owner": "jvansanten", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20160627T20:45:08: claudio.kopper commented:

> from the discussion on Slack:
> 
> 
> jvs [14:31]  
> I'd vote for taking the length seriously.
> 
> claudiok [14:31]  
> I think it’s a good idea
> 
> [14:31]  
> and I think length is currently unused in cascades anyway
> 
> jvs [14:31]  
> The geant backend could even throw if it finds length 0
> 
> claudiok [14:31]  
> but I still think CMC should output a “cascade segment”
> with the necessary length
> �(although this would mean defining a new particle type)
> 
> jvs [14:32]  
> right now it sets length 0, which is what it means, but no one takes it seriously
> 
> claudiok [14:32]  
> okay..
> 
> jvs [14:33]  
> Actually, the more I think about it the more I like the cascade segment with uniform light output
> 
> claudiok [14:33]  
> it’s the right thing to do I think
> 
> jvs [14:33]  
> Just a super "thick" track
> 
> claudiok [14:33]  
> right
> 
> [14:34]  
> and CMC could mark the LPM losses differntly (assuming they are sampled in a different way somehow?)
> 
> jvs [14:37]  
> We (for the same definition of we) need to find out and then document exactly what the losses are supposed to mean. It probably all in Berhards thesis
> 
>

**Comment by icecube-trac** [comment]
Trac update at 20160721T19:51:23: juancarlos commented:

> I added UseCascadeExtension = False in simprod-scripts/modules/clsim.py.

**Comment by icecube-trac** [comment]
Trac update at 20160722T07:25:29: jvansanten commented:

> cmc doesn't do anything to cascades with energies lower than 1 TeV, so we still need to use cascade extension in general. I propose a variant of Claudio's suggestion:
> - Add a new particle shape, `I3Particle::CascadeSegment`
> - When cmc breaks up a cascade, it should emit particles with shape `CascadeSegment` and set their length to the distance between particles (3 radiation lengths by default)
> - clsim should simulate the longitudinal distribution of particles with shape `Cascade` as it currently does. For particles with shape `CascadeSegment`, it should distribute steps uniformly along the length of the segment.
> 
> When cmc simulates the cascade directly to account for the LPM effect, it still bins the results in radiation lengths (see `I3CascadeSimulation::Deposit(Particle particle)`), so the same treatment is valid no matter which mode is used.

**Comment by icecube-trac** [comment]
Trac update at 20160722T14:51:03: juancarlos commented:

> This sounds reasonable. Created tickets #1778 and #1779 to address this.

**Comment by icecube-trac** [comment]
Trac update at 20161003T13:06:51: 
* jvansanten changed owner from "juancarlos" to "jvansanten"
* jvansanten changed status from "new" to "accepted"

**Comment by icecube-trac** [comment]
Trac update at 20161004T14:44:28: 
* jvansanten changed status from "accepted" to "closed"
* jvansanten changed resolution from "" to "fixed"
* jvansanten commented:

> In changeset 150508:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="150508"
> Factor cascade development calcualtion out of clsim into sim-services, and use the same calculation from cmc.Fixes #1875 and #1755.
> 
> This leads to a few side effects:
> - clsim treats cascade segments from cmc properly, without trying to re-apply the cascade extension
> - cmc now uses a different set of shower parameters for EM and hadronic showers
> - cmc uses the correct radiation length in ice
> - clsim simulates hadronic shower extension correctly (the shower scale mixed up hadronic interaction length units and EM interaction length units in r149069
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:12:47: 
* olivas commented:

> Milestone June 2016 deleted


* olivas changed milestone from "June 2016" to ""
